src/Adapter/Module/Repository/ModuleRepository.php

```diff
@@ -24,72 +24,74 @@
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
  */
 
-namespace PrestaShopBundle\Kernel;
+declare(strict_types=1);
 
-use Doctrine\DBAL\Connection;
+namespace PrestaShop\PrestaShop\Adapter\Module\Repository;
+
+use Module;
+use PrestaShop\PrestaShop\Adapter\AbstractObjectModelRepository;
+use PrestaShop\PrestaShop\Core\Domain\Module\Exception\ModuleNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Module\ValueObject\ModuleId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
 use Symfony\Component\Finder\Finder;
 
 /**
- * Before booting the PrestaShop application in Symfony context,
- * we register every installed modules.
+ * Methods to access data source of Module
  */
-final class ModuleRepository
+class ModuleRepository extends AbstractObjectModelRepository
 {
-    /**
-     * @var Connection
-     */
-    private $connection;
-
-    /**
-     * @var string the `modules` table name
-     */
-    private $tableName;
-
     /**
      * @var array
      */
-    private $activeModules;
+    private $activeModulesPaths;
 
     /**
-     * @var array
+     * @param ModuleId $moduleId
+     *
+     * @throws CoreException
+     * @throws ModuleNotFoundException
      */
-    private $activeModulesPaths;
-
-    public function __construct(Connection $connection, $databasePrefix)
+    public function assertModuleExists(ModuleId $moduleId): void
     {
-        $this->connection = $connection;
-        $this->tableName = $databasePrefix . 'module';
+        $this->assertObjectModelExists($moduleId->getValue(), 'module', ModuleNotFoundException::class);
     }
 
     /**
-     * @return array the list of installed modules
+     * @return array
      */
-    public function getActiveModules()
+    public function getActiveModules(): array
     {
-        if (null === $this->activeModules) {
-            $sth = $this->connection->query('SELECT name FROM ' . $this->tableName . ' WHERE active = 1');
-
-            $this->activeModules = $sth->fetchAll(\PDO::FETCH_COLUMN);
+        $activeModules = [];
+        try {
+            $modulesData = Module::getActiveModules();
```

Comment:
the `getActiveModules` request have to be in the repository and not the objectmodel